src/map/utils/battleutils.cpp

```diff
@@ -7087,4 +7088,43 @@ namespace battleutils
         return 1.0;
     }
 
+    bool TryProcTreasureHunter(CCharEntity* PChar, CMobEntity* PMob, CAttackRound* attackRound, bool feintApplied)
+    {
+        // The Treasure Hunter proc system was introduced in the December 2010 update and applies to players using Thief main job, as well as the Ranger job ability Bounty Shot.
+        // https://www.bg-wiki.com/ffxi/Treasure_Hunter
+        // https://ffxiclopedia.fandom.com/wiki/Treasure_Hunter
+
+        auto playerTH = PChar->getMod(Mod::TREASURE_HUNTER);
+        auto mobTH    = PMob->PEnmityContainer->GetHighestTH();
+
+        // Treasure Hunter can be increased up to a maximum value of 12
+        // (boostable by Job Points)
+        auto thCap = 12 + PChar->getMod(Mod::TREASURE_HUNTER_CAP);
+
+        // The maximum base Treasure Hunter level that can be applied from the first action against an enemy is 8.
+        if (!mobTH)
+        {
+            thCap = 8;
+        }
```

Comment:
Doesn't look like this is used yet, but just to note the base cap of 8 is for non-procced hits and _can_ land a 9 on the first hit with a proc.

Reply:
I still don't fully understand the flow here, so regardless of whether you proc or not, your actions will set TH8 on an enemy? and then on top of that you have a chance to proc straight to TH9?

Reply:
This'll also need a bit of a rewrite to track current TH in a way that isn't "for every entity in my enmity list, find the highest TH amount"

Reply:
> I still don't fully understand the flow here, so regardless of whether you proc or not, your actions will set TH8 on an enemy? and then on top of that you have a chance to proc straight to TH9?

pretty much even your very 1st hit has the chance to proc. your base of 8 would then pop the message for 9 on that 1st proc, while raising mob to 9. If you base is 8, you never get a message for 8 because 1st proc brings it to 9.